### ISSUE TITLE
Inconsistent/wrong query behaviour in distributed setup when using aggregate functions

### ISSUE THREAD

**Issue by mcaramma** [orientechnologies/orientdb#7349]
## Expected behavior  
Executing aggregate functions over an empty result set should return an empty result.

For example executing the following query over an empty table should return 0 items found .... in non-distributed setup the behaviour is the expected:
```
orientdb {db=TestDB}> create class Test

Class created successfully. Total classes in database now: 12.

orientdb {db=TestDB}> select sum(fieldA) as fieldA from Test where fieldB = true

0 item(s) found. Query executed in 0.003 sec(s).
```
Executing from Studio:
```
{
    "result": [
        
    ],
    "notification": "Query executed in 0.067 sec. Returned 0 record(s)"
}
```

## Actual behavior  
But in distributed setup the same query surprisingly return 1 item.

## Steps to reproduce  
```
orientdb {db=TestDB}> create class Test

Class created successfully. Total classes in database now: 12.

orientdb {db=TestDB}> select sum(fieldA) as fieldA from Test where fieldB = true

+----+------+
|#   |fieldA|
+----+------+
|0   |      |
+----+------+

1 item(s) found. Query executed in 0.001 sec(s).
```

Executing from Studio return this:
```
{
    "result": [
        {
            "@type": "d",
            "@version": 0,
            "sum": null
        }
    ],
    "notification": "Query executed in 0.061 sec. Returned 1 record(s)"
}
```

The same happen using max() and set() .... I didn't try other functions but I guess the problem is generalized.

**Comment by mcaramma** [comment]
After some more testing it appears that this behaviour only occurs with custered setup, no matters the number of nodes ... verified in a cluster with 3 nodes and also in a cluster with only 1 node.

Also verified that the same incorrect behaviour happens with v2.2.18.

**Comment by mcaramma** [comment]
As an additional note, the same problem happen in v2.2.13 (I didn't try older versions), the difference is that in v2.2.13 functions like set() and list() throw the following error:
```
com.orientechnologies.orient.core.exception.OStorageException: Cannot route COMMAND operation to the distributed node DB name="TestDB" --> java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
```
But that error was fixed in v2.2.14.

**Comment by lvca** [comment]
@mcaramma is the issue present in last 2.2.20 too?

**Comment by mcaramma** [comment]
Yes, it is still present in v2.2.20 ... even worse, now the incorrect behaviour is present in both distributed and non-distributed setup.

```
OrientDB console v.2.2.20 (build 76ab59e72943d0ba196188ed100c882be4315139) https://www.orientdb.com
Type 'help' to display all the supported commands.
Installing extensions for GREMLIN language v.2.6.0

orientdb> connect remote:localhost/Test admin admin                

Connecting to database [remote:localhost/Test] with user 'admin'...OK

CONFIGURED SERVERS
+----+-----+------+-----------+-------------------+----------------+----------------+-----------------+---------------+---------+
|#   |Name |Status|Connections|StartedOn          |Binary          |HTTP            |UsedMemory       |FreeMemory     |MaxMemory|
+----+-----+------+-----------+-------------------+----------------+----------------+-----------------+---------------+---------+
|0   |node1|ONLINE|3          |2017-05-16 12:01:54|10.99.77.90:2424|10.99.77.90:2480|527.26MB (13.43%)|3.32GB (86.57%)|3.83GB   |
+----+-----+------+-----------+-------------------+----------------+----------------+-----------------+---------------+---------+
orientdb {db=Test}> create class Test

Class created successfully. Total classes in database now: 12.

orientdb {db=Test}> select from Test                                  


0 item(s) found. Query executed in 0.002 sec(s).
orientdb {db=Test}> select sum(fieldA) as fieldA from Test where fieldB = 'A'

+----+
|#   |
+----+
|0   |
+----+

1 item(s) found. Query executed in 0.002 sec(s).
```

**Comment by luigidellaquila** [comment]
Hi @mcaramma 

This issue was caused by a fix we made in v 2.2.20. In previous versions, records containing only null values were (wrongly) discarded, now they are correctly returned. 

There was also a bug with sum() function, that returned `null` instead of zero in case no values were summed.
We just pushed a fix for this bug as well, it will be released in 2.2.21.

To recap: the correct behavior is to have, as a result, a single record with fieldA = `0`.

Thanks

Luigi
